test/src/org/exist/xquery/XQueryTest.java

```diff
@@ -1346,6 +1346,17 @@ public void instanceOfDocumentNode() throws XMLDBException {
     }
 
     @Test
+    public void instanceOfNamespaceNode() throws XMLDBException {
+        ResourceSet result = existEmbeddedServer.executeQuery("namespace test { 'test' } instance of namespace-node()");
+        assertEquals(1,result.getSize());
+        assertEquals("true", result.getResource(0).getContent().toString());
+
+        result = existEmbeddedServer.executeQuery("<x/> instance of namespace-node()");
```

Comment:
@shabanovd I am not sure how to test that, or why it is required. I just followed the existing examples. TBH the eXist Antlr 2 grammar is far from clear to me